pkg/storaged/stratis/pool.jsx

```diff
@@ -64,6 +65,197 @@ const _ = cockpit.gettext;
+
+    return result;
+}
+
+function get_clevis_infos(pool) {
+    const result = [];
+
+    if (!pool.Encrypted)
+        return result;
+
+    if (client.stratis_interface_revision < 8) {
+        const val = pool.ClevisInfo;
+        if (val[0] && val[1][0]) {
+            if (val[1][1][0] == "tang") {
+                const config = JSON.parse(val[1][1][1]);
+                result.push({ slot: null, pin: "tang", url: config.url });
+            } else {
+                result.push({ slot: null, pin: val[1][1][0] });
+            }
+        }
+    } else {
+        let val = pool.ClevisInfos;
+        if ("t" in val)
+            val = val.v; // XXX - change notifications drop the variant wrapping for some reason.
+        if (pool.MetadataVersion == 1) {
+            if (val[0] && val[1][0]) {
+                if (val[1][1][0] == "tang") {
+                    const config = JSON.parse(val[1][1][1]);
+                    result.push({ slot: null, pin: "tang", url: config.url });
+                } else {
+                    result.push({ slot: null, pin: val[1][1][0] });
+                }
+            }
+        } else if (pool.MetadataVersion == 2) {
+            for (const ci of val) {
+                if (ci[1][0] == "tang") {
+                    const config = JSON.parse(ci[1][1]);
+                    result.push({ slot: ci[0], pin: "tang", url: config.url });
+                } else {
+                    result.push({ slot: ci[0], pin: ci[1][0] });
+                }
+            }
+        }
+    }
+
+    return result;
+}
+
+function bind_keyring(pool, keydesc) {
+    if (client.stratis_interface_revision < 8)
+        return pool.BindKeyring(keydesc);
+    else
+        return pool.BindKeyring(keydesc, [false, 0]);
+}
+
+function rebind_keyring(pool, keydesc, slot) {
+    if (client.stratis_interface_revision < 8)
+        return pool.RebindKeyring(keydesc);
+    else
+        return pool.RebindKeyring(keydesc, [slot !== null, slot || 0]);
+}
+
+function unbind_keyring(pool, slot) {
+    if (client.stratis_interface_revision < 8)
+        return pool.UnbindKeyring();
+    else
+        return pool.UnbindKeyring([slot !== null, slot || 0]);
+}
+
+function bind_clevis(pool, pin, config) {
+    if (client.stratis_interface_revision < 8)
+        return pool.BindClevis(pin, config);
+    else
+        return pool.BindClevis(pin, config, [false, 0]);
+}
+
+function unbind_clevis(pool, slot) {
+    if (client.stratis_interface_revision < 8)
+        return pool.UnbindClevis();
+    else
+        return pool.UnbindClevis([slot !== null, slot || 0]);
+}
+
+/* Utilities for key descriptions and passphrases
+ */
+
+export async function get_stored_keydescs() {
+    return await client.stratis_manager.ListKeys().catch(() => []);
+}
+
+async function get_new_keydesc(pool) {
+    const key_descs = get_key_descriptions(pool);
+    const stored_keydescs = await get_stored_keydescs();
+
+    let desc;
+    for (let i = 0; i < 1000; i++) {
```

Comment:
Arbitrary 1000 declared here. Is it a limit in Stratis? Should have a comment explaining why as it's static.